### PR TITLE
[MOB-2921] update commerceitem to allow optional fields

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 JIRA Ticket(s) if any

* [MOB-2921](https://iterable.atlassian.net/browse/MOB-2921)

## ✏️ Description

* extends the functionality of `trackPurchase` to match the current API

### CODE REVIEW

File: swift-sdk/CommerceItem.swift [hunk 1]
```diff
@@ -4,49 +4,100 @@
...
+    ///     - name: name of the product
+    ///     - price: price of the product
+    ///     - quantity: quantity of the product
+    ///     - sku: SKU of the eproduct
+    ///     - description: description of the product
+    ///     - url: URL of the product
+    ///     - imageUrl: URL of the product's image
+    ///     - categories: categories this product belongs to
+    ///
+    /// - returns: an instance of `CommerceItem` with the specified properties
```
Comment (tapashmajumder) [inline swift-sdk/CommerceItem.swift:51]: I like this style of comments.

File: swift-sdk/Internal/RequestCreator.swift [hunk 2]
```diff
@@ -89,12 +89,8 @@ struct RequestCreator {
             ITBError("Both email and userId are nil")
             return .failure(IterableError.general(description: "Both email and userId are nil"))
         }
-
-        var itemsToSerialize = [[AnyHashable: Any]]()
         
-        for item in items {
-            itemsToSerialize.append(item.toDictionary())
-        }
+        let itemsToSerialize = items.map { $0.toDictionary() }
```
Comment (tapashmajumder) [inline swift-sdk/Internal/RequestCreator.swift:93]: This is better.

**Comment by tapashmajumder** [comment]
Conditional +1. Please re-run integration tests and make sure nothing is broken there.

**Review by tapashmajumder — APPROVED** [review]
+1. 
Please double check integration tests before merging.

**Comment by roninopf** [comment]
> +1.
> Please double check integration tests before merging.

Looks like we've got some flakiness, it passed after a retry. :/

**Pull request merged**